Add guacscanner

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies this Ansible role to allow for the addition of [cisagov/guacscanner-docker](https://github.com/cisagov/guacscanner-docker) to [cisagov/guacamole-composition](https://github.com/cisagov/guacamole-composition).

## 💭 Motivation and context ##

The functionality of [cisagov/guacscanner-docker](https://github.com/cisagov/guacscanner-docker) is necessary for the RPT UAT taking place in the COOL.

See also:
- cisagov/guacscanner/pull/1
- cisagov/guacscanner-docker#2
- cisagov/guacamole-composition#30
- cisagov/guacamole-packer#53
- cisagov/cool-assessment-terraform#149

## 🧪 Testing ##

I have applied these changes to env0 in our staging COOL environment and verified that they function as expected.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [x] Revert back to the main branch of [cisagov/guacamole-composition](https://github.com/cisagov/guacamole-composition) once cisagov/guacamole-composition#30 is merged.

jsf9k committed:
Add cisagov/guacscanner to the list of Docker images that are pulled down

jsf9k committed:
Temporarily use a particular branch of cisagov/guacamole-composition for testing

jsf9k committed:
Remove defaults

jsf9k committed:
Add more secrets to support guacscanner

jsf9k committed:
Add role variables to README

jsf9k committed:
Add required role variables to converge playbook

jsf9k committed:
Move tests up one directory

This is in preparation for creating multiple scenarios.  GitHub Actions does not provide enough disk space to run all our OSs, so I need to break up the molecule testing.

jsf9k committed:
Split the default scenario

This is necessary because GitHub Actions does not provide enough
disk space to run all our OSs together.

dav3r commented on README.md:
```diff
@@ -12,14 +12,15 @@ None.
+| private_ssh_key | The private ssh key to use for SFTP file transfer in Guacamole. | n/a | Yes |
+| rdp_username | The username for Guacamole to use when connecting to an instance via RDP. | n/a | Yes |
+| rdp_password | The password for Guacamole to use when connecting to an instance via RDP. | n/a | Yes |
+| rdp_username | The username for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
+| rdp_password | The password for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
```
I assume this is what you meant here:
```suggestion
| vnc_username | The username for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
| vnc_password | The password for Guacamole to use when connecting to an instance via VNC. | n/a | Yes |
```

jsf9k committed:
Fix typo

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Use version 13 of PostgreSQL

The version of the JDBC PostgreSQL driver included in the Guacamole
Docker image is too old to work with PostgreSQL 14 or later.

dav3r reviewed (approved):
👍 👍

jsf9k committed:
Improve comment

mcdonnnj commented on molecule/debian/molecule.yml:
```diff
@@ -0,0 +1,67 @@
+  name: ansible
+  inventory:
+    host_vars:
+      debian9_systemd:
```
If we're not testing against Debian 9 (Stretch) is there any reason to keep all of this?

jsf9k replied:
The intent is to remove the Debian 9 jazz wholesale once CyHy moves on from the dark ages.  I can remove it from this particular repo.

jsf9k replied:
Please see commit 57517ab.

mcdonnnj commented on molecule/kali/molecule.yml:
```diff
@@ -0,0 +1,60 @@
+  name: ansible
+  inventory:
+    host_vars:
+      debian9_systemd:
```
Same question about Debian 9 (Stretch).

jsf9k replied:
Please see commit 57517ab.

mcdonnnj commented on molecule/ubuntu/molecule.yml:
```diff
@@ -0,0 +1,67 @@
+  name: ansible
+  inventory:
+    host_vars:
+      debian9_systemd:
```
Third time's the charm?

jsf9k replied:
Please see commit 57517ab.

mcdonnnj reviewed (approved):
Excellent work. I had a question about Debian 9-fu since it is not a supported platform for this role. I also went a bit too far down the rabbit hole and found that [the Dockerfile for guacamole/guacamole](https://github.com/apache/guacamole-client/blob/62b26fb594aa160478ce44c6d2bb8e6a1cef4174/guacamole-docker/bin/build-guacamole.sh#L111) is installing a PostgreSQL JDBC driver that's [over six years old](https://github.com/pgjdbc/pgjdbc/releases/tag/REL9_4_1201). We should probably make an issue in the [Apache JIRA](https://issues.apache.org/jira/projects/GUACAMOLE) to either fix it ourselves with a PR or at least get some attention on it I guess. The closest issue I could find was someone else having a problem in [GUACAMOLE-1433](https://issues.apache.org/jira/browse/GUACAMOLE-1433) and another user mentions the JDBC driver needing an update.

jsf9k committed:
Remove some foo specific to Debian 9

Debian 9 is not a supported platform for this Ansible role.

jsf9k commented:
> I also went a bit too far down the rabbit hole and found that [the Dockerfile for guacamole/guacamole](https://github.com/apache/guacamole-client/blob/62b26fb594aa160478ce44c6d2bb8e6a1cef4174/guacamole-docker/bin/build-guacamole.sh#L111) is installing a PostgreSQL JDBC driver that's [over six years old](https://github.com/pgjdbc/pgjdbc/releases/tag/REL9_4_1201). We should probably make an issue in the [Apache JIRA](https://issues.apache.org/jira/projects/GUACAMOLE) to either fix it ourselves with a PR or at least get some attention on it I guess. The closest issue I could find was someone else having a problem in [GUACAMOLE-1433](https://issues.apache.org/jira/browse/GUACAMOLE-1433) and another user mentions the JDBC driver needing an update.

Good idea.  The old driver has already bitten us.  See cisagov/guacamole-composition@8197d0e804a145a8dac67078b1e0a14c68e8e99d for more details.

jsf9k committed:
Revert "Temporarily use a particular branch of cisagov/guacamole-composition for testing"

This reverts commit 19860a4ae0a5ed1b553b359f5077c8957d0fffde.
